mux.go

```diff
@@ -44,66 +44,51 @@ func Group(s *Server, path string) *Server {
 }
 
 type Route[ResponseBody any, RequestBody any] struct {
-	Operation *openapi3.Operation // GENERATED OpenAPI operation, do not set manually in Register function. You can change it after the route is registered.
-	Method    string              // HTTP method (GET, POST, PUT, PATCH, DELETE)
-	Path      string              // URL path. Will be prefixed by the base path of the server and the group path if any
-	Handler   http.Handler        // handler executed for this route
-	FullName  string              // namespace and name of the function to execute
+	BaseRoute
+}
+
+type BaseRoute struct {
+	Operation   *openapi3.Operation // GENERATED OpenAPI operation, do not set manually in Register function. You can change it after the route is registered.
+	Method      string              // HTTP method (GET, POST, PUT, PATCH, DELETE)
+	Path        string              // URL path. Will be prefixed by the base path of the server and the group path if any
+	Handler     http.Handler        // handler executed for this route
+	FullName    string              // namespace and name of the function to execute
+	Params      map[string]OpenAPIParam
+	Middlewares []func(http.Handler) http.Handler
 
 	mainRouter *Server // ref to the main router, used to register the route in the OpenAPI spec
 }
 
 // Capture all methods (GET, POST, PUT, PATCH, DELETE) and register a controller.
-func All[T, B any, Contexted ctx[B]](s *Server, path string, controller func(Contexted) (T, error), middlewares ...func(http.Handler) http.Handler) Route[T, B] {
-	return Register(s, Route[T, B]{
-		Path:     path,
-		FullName: FuncName(controller),
-	}, HTTPHandler(s, controller), middlewares...)
+func All[ReturnType, Body any, Contexted ctx[Body]](s *Server, path string, controller func(Contexted) (ReturnType, error), options ...func(*BaseRoute)) *Route[ReturnType, Body] {
```

Comment:
Just to clarify for myself. The only way to add middlewares to a specific route now is to use `options`

Reply:
For route-scoped middlewares, yes.

Using `option.Middleware(myMiddleware)`

Reply:
👍